### PR TITLE
Add aptardi

### PR DESCRIPTION
**Pull request by luskry**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by luskry** [comment]
Updated meta.yaml to prevent test error.

**Comment by luskry** [comment]
@BiocondaBot please add label

File: recipes/aptardi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+requirements:
+  host:
+    - python==3.7.7
+  run:
+    - python==3.7.7
+    - tensorflow==2.0.0
+    - numpy==1.18.1
+    - pandas==1.0.3
+    - scikit-learn==0.22.2.post1
```
Comment (dpryan79) [inline recipes/aptardi/meta.yaml:21]: This only works with these exact versions? So not python 3.8 or 3.7.6?

**Comment by luskry** [comment]
It should work on all python 3.7 versions. It doesn’t work on 3.8 just because it only works on tf 2.0.0 and this version of tf isn’t supported on 3.8 and up versions of python.

On May 20, 2021, at 12:18 AM, Devon Ryan ***@***.***> wrote:

﻿

@dpryan79 commented on this pull request.

________________________________

In recipes/aptardi/meta.yaml<https://github.com/bioconda/bioconda-recipes/pull/28583#discussion_r635771953>:

> +  url: https://github.com/luskry/aptardi/archive/refs/tags/v.1.3.13.tar.gz
+  sha256: 4ab792eff02606c37c9bcdebba94eb10a1ff5d756c9097089d8bcd54e6dbaca8
+
+build:
+  noarch: python
+  number: 0
+
+requirements:
+  host:
+    - python==3.7.7
+  run:
+    - python==3.7.7
+    - tensorflow==2.0.0
+    - numpy==1.18.1
+    - pandas==1.0.3
+    - scikit-learn==0.22.2.post1


This only works with these exact versions? So not python 3.8 or 3.7.6?

**Comment by dlaehnemann** [comment]
The idea is generally to unpin as many dependencies as possible. Otherwise, it may get really hard to install aptardi in an environment with other tools, and particular versions of the dependencies might even disappear. So some flexibility is usually best.

You could start with `- python >=3.7,<3.8` and also look into looser pinning for everything except for `tensorflow`, which seems to be a strict requirement to be pinned to this particular version? Or am I misunderstanding you?

For more details on version specifications for conda packages, see here:
https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications

**Comment by luskry** [comment]
Yes, you understand that correctly. TensorFlow versions greater than 2.0.0 create a bug when trying to do cost sensitive learning due to the imbalanced classes. I am working on eventually getting this to work with newer versions of TensorFlow.

I updated the meta.yaml to allow for less strict package dependencies. I remember there was also an issue with scikit-learn v.0.23.2, but it may be okay with the v.0.24 and up releases. I am not 100% sure on this though. I have the version right now as >=v.0.24.
________________________________

**Comment by dlaehnemann** [comment]
Once tests have run through, you can also try out the resulting packages locally. If you click on the triangle next to `packages` in the post by the BiocondaBot once it is done with the latest update, it gives you instructions on how to use them. So if you have good test cases on your machine, you can try it on them.

**Comment by luskry** [comment]
Yea that's what I was thinking I have some data to prototype so I'll check it out.
________________________________

**Comment by luskry** [comment]
Encountered an issue loading the keras model, trying with a specific version h5py based on stack overflow recommendation.

**Comment by luskry** [comment]
@dlaehnemann @dpryan79 Ok so I updated the meta.yaml to allow more flexibility with packages. Allowing flexibility with python require numpy<=1.19.5 and h5py<=2.10.0. Is this okay?

File: recipes/aptardi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package:
+  name: aptardi
+  version: '1.4'
+
+source:
+  url: https://github.com/luskry/aptardi/archive/refs/tags/v.1.4.tar.gz
+  sha256: e7cae7e940573ffdb7db90cb856fdd38824e077ce0df88e95f3bf6a03ba61d43
```
Comment (dlaehnemann) [inline recipes/aptardi/meta.yaml:7]: It's good practice to use jinja variable wherever possible, so that you only have to update things in one place:

```suggestion
{% set name = "aptardi" %}
{% set version = "1.4" %}
{% set sha256 = "e7cae7e940573ffdb7db90cb856fdd38824e077ce0df88e95f3bf6a03ba61d43" %}

package:
  name: {{ name }}
  version: {{ version }}

source:
  url: https://github.com/luskry/{{ name }}/archive/refs/tags/v.{{ version }}.tar.gz
  sha256: {{ sha256 }}
```

Also, I think this makes it easier for the BiocondaBot to auto-update the recipe as soon as a new release appears on the repo.

File: recipes/aptardi/build.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+$PYTHON setup.py install
```
Comment (dlaehnemann) [inline recipes/aptardi/build.sh:3]: I would suggest deleting this file altogether, in favor of the `build: script:` directive in `meta.yaml`.
```suggestion
```

**Review by dlaehnemann — CHANGES_REQUESTED** [review]
The pinning looks much better, thanks a bunch.

I added some more suggestions that should make the recipe easier to maintain.

**Review by luskry — COMMENTED** [review]
Sounds great - thanks for your help!

**Review by dlaehnemann — APPROVED** [review]
Thanks for bearing with us! Looks good, now.
